elasticdl/python/tests/worker_pytorch_training_loop_test.py

```diff
@@ -0,0 +1,178 @@
+                for (x, y) in test_db:
+                    x = torch.from_numpy(x.numpy()).type(torch.FloatTensor)
+                    y = torch.from_numpy(y.numpy()).type(torch.LongTensor)
+                    x = torch.unsqueeze(x, dim=1)
+
+                    out = worker._model.forward(x)
+                    if "mnist" in self._model_def:
+                        acc_meter.update_state(
+                            torch.argmax(out, dim=1).numpy(), y.numpy()
+                        )
+                    else:
+                        break
+                worker_results.append(
+                    (
+                        float(w_loss.detach().numpy()),
+                        float(acc_meter.result().numpy()),
+                    )
+                )
+                acc_meter.reset_states()
+
+            if step > stop_step:
+                break
+        return worker_results
+
+    def test_mnist_train(self):
+        model_def = "mnist.mnist_subclass_pytorch.CustomModel"
+        self._create_pserver(model_def, 2)
+
+        # Reuse the tf style code to get dataset
+        db, test_db = get_mnist_dataset(self._batch_size)
+
+        stop_step = 20
+        self._create_worker(1)
+        worker_results = self._worker_train(
+            0, train_db=db, test_db=test_db, stop_step=stop_step
+        )
+        print(
+            "show batchs train worker_results[0]\n" "w_loss, acc_meter:",
+            worker_results[0],
+        )
+        print("--------finish test_compare_mnist_train!")
+
+    def test_onebatch_train_pytorch(self):
+        model_def = "mnist.mnist_subclass_pytorch.CustomModel"
+        self._create_pserver(model_def, 2)
+
+        images, labels = get_random_batch(self._batch_size)
+        images = torch.unsqueeze(
+            torch.from_numpy(images.numpy()), dim=1
+        ).float()
+        labels = torch.from_numpy(labels.numpy()).type(torch.int64)
+        # Focus: labels.dtype is torch.Long/torch.int64
+
+        # TODO(yunjian.lmh): test optimizer wrapper
+        arguments = [
+            "--worker_id",
+            0,
+            "--job_type",
+            elasticdl_pb2.TRAINING,
+            "--minibatch_size",
+            self._batch_size,
+            "--model_zoo",
+            self._model_zoo_path,
+            "--model_def",
+            model_def,
+            "--distribution_strategy",
+            DistributionStrategy.PARAMETER_SERVER,
+        ]
+        args = parse_worker_args(arguments)
+        worker = WorkerPytorch(args, ps_client=PSClient(self._channels))
+        worker._run_model_call_before_training(images)
+        worker.get_model()
+        w_loss, w_grads = worker.training_process_pytorch(images, labels)
+        worker.report_gradient(w_grads)
+        print("show onebatch train w_loss:", w_loss)
+        print("show onebatch train w_grads[0]:", w_grads[0].shape)
+        print("--------finish test_compare_onebatch_train_pytorch!")
```

Comment:
Delete `print` and you can use assert like `self.assertEqual` to check the test.